src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1171,17 +1171,19 @@ private void CleanUp()
 
             try
             {
-                // on Unix, we need to kill the process to ensure it terminates as Dispose() merely
-                // closes the redirected streams and the processs does not exit on macOS.  However,
-                // on Windows, a winexe like notepad should continue running so we don't want to kill it.
+                // on Unix, we need to kill the process (if not running in background) to ensure it terminates,
+                // as Dispose() merely closes the redirected streams and the process does not exit.
+                // However, on Windows, a winexe like notepad should continue running so we don't want to kill it.
```

Comment:
I don't see an issue here since `Complete` is only for when the process runs in the foreground and the existing code was already there to explicitly `Dispose`

Reply:
I mean, when exception happens in `Prepare` or `ProcessRecord`, we may want to kill the process on Unix platforms even if it's running in background.

Reply:
I see, let me review that

Reply:
looks like this should be the case that in `Prepare` and `ProcessRecord` if something raises an exception, then we should cleanup even background processes